src/server/dcp_server/models.py

```diff
@@ -60,12 +70,444 @@ def masks_to_outlines(self, mask):
+            Input:
+            img: np.ndarray[np.uint8]
+            Output: y_hat - The predicted label
+        """ 
+        # normalise
+        img = (img-np.min(img))/(np.max(img)-np.min(img))
+        # convert to tensor
+        img = torch.permute(torch.tensor(img.astype(np.float32)), (2, 0, 1)).unsqueeze(0) 
+        preds = self.forward(img)
+        y_hat = torch.argmax(preds, 1)
+        return y_hat
+
+class CellposePatchCNN():
+
+    """
+    Cellpose & patches of cells and then cnn to classify each patch
+    """
+    
+    def __init__(self, model_config, train_config, eval_config):
+        
+        self.model_config = model_config
+        self.train_config = train_config
+        self.eval_config = eval_config
+
+        # Initialize the cellpose model and the classifier
+        self.segmentor = CustomCellposeModel(self.model_config["segmentor"], 
+                                             self.train_config["segmentor"],
+                                             self.eval_config["segmentor"])
+        self.classifier = CellClassifierFCNN(self.model_config["classifier"],
+                                             self.train_config["classifier"],
+                                             self.eval_config["classifier"])
+
+    def init_from_checkpoints(self, chpt_classifier=None, chpt_segmentor=None):
+        """
+        Initialize the model from pre-trained checkpoints.
+        """
+
+        self.segmentor = CustomCellposeModel(
+            model_config={"gpu":torch.cuda.is_available(), "pretrained_model":chpt_segmentor} 
+            )
+        self.classifier.load_state_dict(torch.load(chpt_classifier)["model"])
+
 
+    def train(self, imgs, masks):
+        # masks should have first channel as a cellpose mask and all other layers
+        # correspond to the classes, to prepare imgs and masks for training CNN
+
+        # TODO I commented below lines. I think we should expect masks to have same size as output of eval, i.e. one channel instances, second channel classes
+        # in this case the shape of masks shall be [2, H, W] or [2, 3, H, W] for 3D
+        # In this case remove commented lines
+        # num_classes = self.model_config["classifier"]["num_classes"]
+        # masks_1channel = [mask.sum(0) for mask in masks]
+        # masks_classifier = [mask if mask.shape[-1] == num_classes else
+                                 # mask.transpose(1, 2, 0) for mask in masks]
+        
+        # train cellpose
+        masks = np.array(masks)
+        masks_instances = list(masks[:,0, ...])
+        self.segmentor.train(imgs, masks_instances)
+        # create patch dataset to train classifier
+        masks_classes = list(masks[:,1, ...])
+        patches, labels = self.create_patch_dataset(imgs, masks_classes, masks_instances)
+        # train classifier
+        self.classifier.train(patches, labels)
+        #return # TODO - define if we need to return something
```

Comment:
remove - nothing should be returned from train (see train of CustomCellposeModel)